web-api/src/business/useCaseHelper/automaticBlock/updateCaseAutomaticBlock.ts

```diff
@@ -3,30 +3,30 @@ import { ServerApplicationContext } from '@web-api/applicationContext';
 import { createCaseTrialSortMappingRecords } from '@web-api/persistence/dynamo/cases/createCaseTrialSortMappingRecords';
 import { deleteCaseTrialSortMappingRecords } from '@web-api/persistence/dynamo/cases/deleteCaseTrialSortMappingRecords';
 import { getCaseDeadlinesByDocketNumber } from '@web-api/persistence/postgres/caseDeadlines/getCaseDeadlinesByDocketNumber';
+import { isEmpty } from 'lodash';
 
-/**
- * updateCaseAutomaticBlock
- *
- * @param {object} providers the providers object
- * @param {object} providers.applicationContext the application context
- * @param {object} providers.caseEntity the case entity to update
- * @returns {object} the updated case entity
- */
 export const updateCaseAutomaticBlock = async ({
   applicationContext,
   caseEntity,
+  hasCaseDeadline,
 }: {
   applicationContext: ServerApplicationContext;
   caseEntity: Case;
+  hasCaseDeadline?: boolean;
 }) => {
   if (caseEntity.trialDate || caseEntity.highPriority) {
     return caseEntity;
   }
-  const caseDeadlines = await getCaseDeadlinesByDocketNumber({
-    docketNumber: caseEntity.docketNumber,
-  });
 
-  caseEntity.updateAutomaticBlocked({ caseDeadlines });
+  if (hasCaseDeadline === undefined) {
```

Comment:
Is it safe to assume that we don't have access to this information in the other places where `updateCaseAutomaticBlock` is called?

Reply:
Good call out. That is correct.